src/cpp/core/http/SocketProxy.cpp

```diff
@@ -182,8 +190,16 @@ void SocketProxy::handleError(const boost::system::error_code& e,
 
 void SocketProxy::close()
 {
-   ptrClient_->close();
-   ptrServer_->close();
+   // ensure we only close the connections once
+   // closing sockets more than once is considered misuses by boost
+   // and can (and has) lead to segfaults
+   if (!closed_)
+   {
+      ptrClient_->close();
```

Comment:
Do you think we should store the `closed` state in the individual `Socket` instances instead? Then we could defend against other code paths which inadvertently double-close a `Socket` (rather than just the `SocketProxy`)